### PR TITLE
LearnUpon

### PR DESCRIPTION
**Pull request by omercnet**
## Status
In Progress

## Related Issues
N/A

## Description
Basic learn upon integration used internally by Demisto

## Required version of Demisto
3.6

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Additional changes
N/A

### CODE REVIEW

File: Integrations/integration-LearnUpon.yml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+description: LearnUpon is a powerful LMS that instantly simplifies online learning
+configuration:
```
[the remark this answers is not in the document]
Comment (omercnet) [reply Integrations/integration-LearnUpon.yml]: my initial thought was that you should never work insecurely with SaaS products but I realize some organizations have proxies that strip SSL so we'll have to allow it. Will add

File: Integrations/integration-LearnUpon.yml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    s = requests.Session()
+
+
+    def request(method, endpoint, data=None):
+        r = s.request(method,
+                      auth=(credentials['identifier'], credentials['password']),
+                      url=urljoin(url, endpoint),
+                      json=data)
+        try:
+            r.raise_for_status()
```
[the remark this answers is not in the document]
Comment (omercnet) [reply Integrations/integration-LearnUpon.yml]: `raise_for_status` will raise on anything that's not 200 - which is what I'm expecting from this API

File: Integrations/integration-LearnUpon.yml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+            demisto.results({'Type': entryTypes['error'], 'Contents': r.json(), 'ContentsFormat': formats['text']})
+            sys.exit(0)
+        return r.json()
+
+
+    def search_user():
+        users = request('GET', 'users/search?email=' + demisto.getArg('email'))
+        md = tableToMarkdown('LearnUpon User', users['user'], ['id', 'first_name', 'last_name', 'user_type']);
+        ec = { 'LearnUpon': { 'Users': users['user'] } }
+        demisto.results({'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': users['user'], 'HumanReadable': md, 'EntryContext': ec})
```
[the remark this answers is not in the document]
Comment (omercnet) [reply Integrations/integration-LearnUpon.yml]: The linter in Demisto doesn't complain, open a feature request for additional rules if you have them :)

File: Integrations/integration-LearnUpon.yml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    def invite_user():
+        data = {'GroupInvite': { 'email_addresses': demisto.getArg('email'), 'group_membership_type_id': user_type[demisto.getArg('user_type')] } }
+
+        res = request('POST', 'group_invites', data)
+        md = tableToMarkdown('LearnUpon Invites', res['group_invites'], ['id', 'invite_email_address']);
+        ec = { 'LearnUpon': { 'InvitedUser': res['group_invites'] } }
+        demisto.results({'ContentsFormat': formats['json'], 'Type': entryTypes['note'], 'Contents': res['group_invites'], 'HumanReadable': md, 'EntryContext': ec})
+
+
+    if demisto.command() == 'test-module':
```
[the remark this answers is not in the document]
Comment (omercnet) [reply Integrations/integration-LearnUpon.yml]: can you link an example?

File: Integrations/integration-LearnUpon.yml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    def list_users():
+        return request('GET', 'users')
+
+
+    def invite_user():
+        data = {'GroupInvite': { 'email_addresses': demisto.getArg('email'), 'group_membership_type_id': user_type[demisto.getArg('user_type')] } }
+
+        res = request('POST', 'group_invites', data)
+        md = tableToMarkdown('LearnUpon Invites', res['group_invites'], ['id', 'invite_email_address']);
+        ec = { 'LearnUpon': { 'InvitedUser': res['group_invites'] } }
```
[the remark this answers is not in the document]
Comment (omercnet) [reply Integrations/integration-LearnUpon.yml]: image

**Comment by omercnet** [comment]
@Itay4 please review

File: Integrations/integration-LearnUpon.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (anara123) [reply Integrations/integration-LearnUpon.yml]: you can read here https://github.com/demisto/content/blob/master/docs/tutorial-video/README.md

File: Integrations/integration-LearnUpon.yml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
...lwdL9IPk4j2SGJEgtfHI4DKM3SGLAH2EBhWCYTo9Ff3ZwWxdivOcmtUJmr3guhr92D28Kwmvfrw6r1bfkgZHCSF0Yi9zhFU8Blbt1fGa500KNEwBOmLAjLbHDAMYq/NoVBLz41W/2jwfmAJ9kpjiaNRJY6NZzH8qaTpno8DczxplDjGt1KsSITALtVNAAdBkFnni3JRurBtQewKbzifYWTskNAE+KaanAc/ojWEF1N8sN8dYor60e8lc316mwenXsDJqRv6/QwFAq3a4/79T9b2gprUgJvHOk8a6QgtWzcNS+T72ksV7rcsbqJDaWrLp85ofSjwjaF46CmDCbZgVkgRQ/XkiuKdXrHNkjsMKksDIGlsdARjfVpx7lZ0WDGf6NawMm5GbFGhSYCRQQAJUbVCnB8h6ls39M65bXs9m2B3WDob1IiLWLG3WvjjwlV6SuV4Pf/K7EBx1HvwF93YBwEiFsqFsc1pQzGU6v40JwvttaQKzhoGL1nzSpECTAns5BXYGMVVIQMY2mcxVs//oFiszWZ6/B1ulqIwteSpM0tK/FI8SnF6NP68MKocM8FQ23IpoywXf6/sgZZJima0Uu6i7sAk0f5oUaFJgb6BA7XSyusSyOkYisx/DZKmGzcbvQ1P1r9qyou/aWQM2GUUOYU9YneRuOdc8FAMi9pPZ5DD2Y1Nm1woMFT/nfe2ZZXJv5UPg2gxNCow8CuAcxSuUBln9+ZdNjt2lthoYxJScnVp6oJBznO9y57uJS1e+iSnmbIwyj+d6f8gf7A7UG69s8NYdmdYWXsS+/W/4+boH2+zbM5+f+5TN2OrzHKbDr2wxmqk3KfCyKBAqMxRbrTA/duXTrqaDZ/bmy0r7/+DLg4OYCCJ7jLSOGGUCarkr593Nlf68KVvnWXI6sWdCx4uOeTGMUzi2EVUHlc0zGNWaQjM0KTCyKSBJzNs+g+wlYYzhuFLWspcnizVuhzWy67mHSjd85LfbDTCIC4wv91BBa7ORAR1WxjbIAG+YhpW1iTWvmhTYUxRQf3PNZ9a0xKLhKSasr04QtBUn4obL+Upn9vPznuKOF88+bP4MRYHhg1htirzvz+dr77/CV02F/StM4GbyTQrsNRR4uSC211S0WdAmBUYQBepv7WLvVHNWMYJaqVmUJgWaFGhSYE9Q4P8DkRKQNHfD/z8AAAAASUVORK5CYII=
```
[the remark this answers is not in the document]
Comment (anara123) [reply Integrations/integration-LearnUpon.yml]: @omercnet change the logo to be 4k

**Comment by omercnet** [comment]
@trorabaugh please take this over

**Comment by Itay4** [comment]
@trorabaugh what's the status of this ?

**Comment by omercnet** [comment]
closing due lack of interest 
@trorabaugh please reopen if you still need it

**Pull request closed without merging**